src/Illuminate/Support/Arr.php

```diff
@@ -294,11 +294,11 @@ public static function get($array, $key, $default = null)
         }
 
         foreach (explode('.', $key) as $segment) {
-            if (static::accessible($array) && static::exists($array, $segment)) {
-                $array = $array[$segment];
-            } else {
-                return value($default);
+            if (!static::accessible($array) || !static::exists($array, $segment)) {
```

Comment:
Why is a change in this condition needed?

Reply:
Cause Previously it was both of them needs to be true. If any of them turns to `false` the condition check will fail. As we are inverting it, so we need to check either one is `false` or not. Hope you get it.

Reply:
I see a few problems with this:

For me is more natural to read "if accessible and exists" as "if not accessible or not exists".

Reply:
Uh... It's ***not a problem*** actually, it's may be your *personal preference* :smiley: Anyway, to me it's more readable than the one with `else`. Also, early `return` is preferable I think and we try to follow it in other parts of *Laravel* code base.

Reply:
Well for me if the element is not accessible there are more chances that it does not exist, this will mean that the second condition will be always evaluated when not needed.

With the previous code, it was clear that after an array is not accessible we will return the default value.

Reply:
By inverting the condition position we can solve this issue @jmarcher :thinking:. See my last commit.